destination/destination.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright © 2022 Meroxa, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package destination
+
+import (
+	"context"
+	"net/http"
+	"sync"
+
+	dConfig "github.com/conduitio/conduit-connector-google-sheets/destination/config"
+	"github.com/conduitio/conduit-connector-google-sheets/destination/writer"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"golang.org/x/oauth2"
+)
+
+type Destination struct {
+	sdk.UnimplementedDestination
+
+	Buffer            []sdk.Record
+	AckCache          []sdk.AckFunc
+	Error             error
+	Token             *oauth2.Token
+	Client            *http.Client
+	Mutex             *sync.Mutex
+	DestinationConfig dConfig.Config
```

Comment:
Let's check which fields actually have to be exported.

Reply:
converted them to internal fields